### ISSUE TITLE
Stream drops frames

### ISSUE THREAD

**Issue by NigNog9001** [mixer/obs-studio-ftl#111]
Done the following;

Reinstalled OBS
Restarted modem/router
Made sure the settings of OBS are the same as the requirement page from Mixer
https://watchbeam.zendesk.com/hc/en-us/articles/115000638643-Streaming-With-OBS-FTL

https://gist.github.com/anonymous/abcffffdb14e991d2093d2edf0603467

**Comment by narcogen** [comment]
Looks to me like the app isn't going to be functional on your system at all due to missing libraries:

```
20:53:33.042: Required module function 'obs_module_load' in module '../../obs-plugins/64bit/chrome_elf.dll' not found, loading of module failed
20:53:33.045: [CoreAudio encoder]: Adding CoreAudio AAC encoder
20:53:33.045: [AMF] Encountered Exception during AMF initialization: Unable to load 'amfrt64.dll', error code 126.
20:53:33.045: Failed to initialize module 'enc-amf.dll'
20:53:33.050: Required module function 'obs_module_load' in module '../../obs-plugins/64bit/libcef.dll' not found, loading of module failed
20:53:33.050: Required module function 'obs_module_load' in module '../../obs-plugins/64bit/libEGL.dll' not found, loading of module failed
20:53:33.050: Required module function 'obs_module_load' in module '../../obs-plugins/64bit/libGLESv2.dll' not found, loading of module failed
```

**Comment by CameronBunch** [comment]
I can second this issue.

For me, I can load the alpha build up, it works fine on the first scene or for the first few minutes but the second I either switch scenes or a movement happens on the beginning scene I start dropping frames.  

What I have done for troubleshooting:
Duplicated issue on both wired and wireless.
Restarted modem/router
Called ISP and I have been told I am green across the board.
I have changed bitrates around between 1800 - 3500 for 720p 60fps and 720p 30fps.
Have tried on both CPU x264 and NVENC.

When I revert to the previous version 18.0.1 with FTL,  I am not seeing these issues.
I have attached the latest log for the alpha build.
[2017-08-04 15-14-17.txt](https://github.com/mixer/obs-studio-ftl/files/1202628/2017-08-04.15-14-17.txt)

**Comment by JimmyMac3** [comment]
I have seen this issue and the high CPU issues before. Long before Alpha OBS-Studio. I believe the problem is OBS-Studio and OBS-FTL share the same settings / profile / scenes file locations. But not necessarily are they compatible with both programs / versions. Switching back and forth between OBS-FTL and OBS-Studio can overwrite the settings files and may be causing  corruption. 

Try this to test this theory. 

1. Make a backup of %appdata%\obs-studio folder including all sub folders (this is your current scenes and settings)
2. Delete the %appdata%\obs-studio folder
3. Run OBS-Studio (Alpha version) and set up all your settings from scratch. 
4. Create a scene so you can test and see if you still have the same issues. 

Be sure that at no time while your testing you run OBS-FTL. If the drop frames stop then you will know how to fix it. Just setup everything from scratch and never run OBS-FTL again.

**Comment by CameronBunch** [comment]
Yes, I thought it might have been the same issue as you described but when I did my testing I always uninstall older OBS versions just to be on the safe side.

So just to be on the safe side I went step by step through what you said.
Uninstalled and removed the %appdata%\obs-folder just in case this time still having the same issues.
I also search through just to be sure no remains of previous builds were around.
Installed only the OBS-Studio (Alpha Build) and completely re-setup all setting by scratch.
Still causing the same results I started with a static scene of an image and then moved to a game scene and immediately moving to the game scene I would drop frames.

I did a complete uninstall of OBS Alpha build, manually removed the %appdata%\obs-studio folder and installed the previous OBS-FTL version 18 and it works 100%.

**Comment by QuinnDamerell-MS** [comment]
Are you still seeing this with the latest release version of OBS?

**Comment by georgeFluff** [comment]
Just saw this, I have this issue and I drop >100% of the frames
![mixer stream kbits](https://user-images.githubusercontent.com/31552555/29988061-9e049b74-8f6b-11e7-9c01-d5bd035f0ddf.PNG)

**Comment by Vaygrim** [comment]
I'm suddenly getting this exact same issue. Latest build of OBS Studio for Windows, as of DEC-22-2017.

**Comment by Maximilian-Reuter** [comment]
If I remember correctly there was a change, that included lost packets in the dropped frame statistic

**Comment by narcogen** [comment]
Should probably report it to the OBS devs, then, and not here.

**Comment by Maximilian-Reuter** [comment]
well thing is it kinda happened here https://github.com/mixer/obs-studio-ftl/blob/c7f8471eadb1af04ff89c0de23e24d363b5a3d80/plugins/obs-outputs/ftl-stream.c#L957 so...

**Comment by QuinnDamerell-MS** [comment]
Yup. We implemented the API wrong and increment the dropped frame count when we lose a single packet. Since frames typically get broken into 6-8 packets, you can get in the hundreds. Also, lost packets aren't the end of the world - we can recover some of them. We will fix this soon. :)
